clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -1226,6 +1203,34 @@ public void run() {
     }
   }
 
+  /**
+   * Create and return the directory for storing RMD schema files of this store.
+   * @return
+   * @throws IOException
+   */
+  String getRmdSchemaPath() throws IOException {
+    // make the base directory TEMP_DIR_PREFIX with 777 permissions
+    Path baseSchemaDir = new Path(TEMP_DIR_PREFIX);
+    FileSystem fs = FileSystem.get(new Configuration());
+    if (!fs.exists(baseSchemaDir)) {
+      fs.mkdirs(baseSchemaDir);
+      fs.setPermission(baseSchemaDir, new FsPermission("777"));
+    }
+
+    // build the full path for HDFSRmdSchemaSource: the schema path will be suffixed
+    // by the store name and time like: <TEMP_DIR_PREFIX>/<store_name>/<timestamp>
+    StringBuilder schemaDirBuilder = new StringBuilder();
+    schemaDirBuilder.append(TEMP_DIR_PREFIX)
+        .append(pushJobSetting.storeName)
+        .append("/")
+        .append(System.currentTimeMillis());
```

Comment:
Do we have logic to clean up this temp folder after repush?
If there is a leaking issue, will next repush will cleanup the leaked files?
Or are we relying on HDFS to do the auto cleanup of temp folder?

Reply:
No, we don't have logics to clean up the tmp folder so I think we are relying on the HDFS (although I haven't confirmed whether they will do the cleanup or not).

That being said, it's good to have some sort cleanup from our side. I will add some codes later on to support always cleaning up the directory for the given store before writing new RMD schemas.

Reply:
-- Correction --

We do have this logic.. I kinda missed this. The schema directory would be cleaned up after each push job [at this line](https://github.com/linkedin/venice/blob/0a128f54da7fe1574acffb57af211bb32b56fd77/clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java#L1066). Not the most neat way of doing it but it's good for now.